src/Symfony/Contracts/HttpClient/HttpClientInterface.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Contracts\HttpClient;
+
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\Test\HttpClientTest;
+
+/**
+ * Provides flexible methods for requesting HTTP resources.
+ *
+ * @see HttpClientTest for a reference test suite
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @experimental in 1.1
+ */
+interface HttpClientInterface
+{
+    public const OPTIONS_DEFAULTS = [
+        'data' => null,         // mixed - any value that MUST be returned as is as the "data" info key of the response
+        'auth' => null,         // string - a username:password enabling HTTP Basic authentication
```

Comment:
Does the `auth` option need to be URL-encoded beforehand? If so, we should mention that

Reply:
I didn't remember that Guzzle uses `['name', 'pass']` instead of `'name:pass'`  Maybe it's a better idea to pass an array like Guzzle?

Reply:
Now throwing if not a string.
Not url-encoded, correct (we're not in a url context so no need to encode).
Using array: I don't know. Guzzle does it to allow more auth schemes than just "Basic" - could be a reason to do the same. Can be considered in future PR.

Reply:
Using `[$user, $pass]` for Basic auth and passing a string as-is would be useful. I'm using `Bearer $token` for authentication more and more now, Basic hardly ever gets used :slightly_frowning_face:

Reply:
For Bearer, you don't really need an option: the header is trivial to declare: `['headers' => 'Authorisation: Bearer '.$myToken]` that's why I did propose it.

Reply:
if it is only for Basic auth, what about renaming the option to `basic_auth` or `auth_basic` ?